setup.py

```diff
@@ -52,6 +52,7 @@
         # TODO: upgrade jupyter-client once
         #  https://github.com/jupyter/jupyter_client/issues/637 is fixed
         "jupyter-client~=6.1.12",
+        "mypy~=0.791",
```

Comment:
So far we've used `==` for things that are in `0.x` version (as under SemVer this allows breaking changes). I'm a bit unsure on the merits of it, but I think we should at least be consistent within the file. Additionally `~=0.791` actually allows anything within `0.x` (there's no major/minor branch) so is quite permissive.

Reply:
Sure. I guess the reason I did that is if mypy adds support for some of the stuff I had to hardcode, we could get it right away. But we would have to remove the `#type ignore` comments / the `cast`s anyways, so you're probably right.